__TESTS__/unit/urlConfig.test.ts

```diff
@@ -0,0 +1,74 @@
+import TransformableImage from "../../src/transformation/TransformableImage";
+import Resize from "../../src/actions/resize/Resize";
+import ICloudinaryConfigurations from "../../src/interfaces/Config/ICloudinaryConfigurations";
+
+const DEMO_CONFIG = {
+  cloud: {
+    cloudName: 'demo'
+    // Other cloud-based properties
+  }
+};
+
+describe('It tests a combination of Cloudianry URL and Configuration', () => {
+  it ('Generates a URL', () => {
+    const url = new TransformableImage('my_image')
+      .setConfig(DEMO_CONFIG)
+      .toURL();
+
+    expect(url).toBe('http://res.cloudinary.com/demo/image/upload/my_image');
+  });
+  it ('Generates a URL with transforamtions', () => {
+    const url = new TransformableImage()
+      .setConfig(DEMO_CONFIG)
+      .resize(Resize.fill(100, 100))
+      .setPublicID('sample')
+      .describeAsset({
+        // has sane defaults for image/upload
+      })
```

Comment:
why do we need that?

Reply:
This is an optional function that allows us to add additional asset information (such as type, resourceType etc.)

You don't have to use it, this is strictly for showing purposes that it exists with some defaults